bin/main.rb

```diff
@@ -1,97 +1,137 @@
 #!/usr/bin/env ruby
+# rubocop: disable Metrics/MethodLength
+require_relative '../lib/game.rb'
+require_relative '../lib/player.rb'
 
-def welcome_message
-  puts '__' * 50
+game = Game.new
+
+def space
   puts ''
+end
+
+def welcome_message
+  puts '_' * 80
+  space
   puts '                      WELCOME TO THE WONDER GAME OF TIC TIC TOE'
-  puts '__' * 50
-  puts ''
+  puts '_' * 80
+  space
   puts 'Players take turns to select from 9 positions on the board below'
   puts 'First to make a horizontal or vertical line of their selection wins'
-  puts ''
+  space
+end
+
+def player_input
+  space
+  space
+  puts 'Please Select Spot 1 - 9'
+  space
+  gets.chomp.to_i - 1
 end
 
-board = %w[1 2 3 4 5 6 7 8 9]
 def display_board(board)
+  space
   puts " #{board[0]} | #{board[1]} | #{board[2]} "
   puts '-----------'
   puts " #{board[3]} | #{board[4]} | #{board[5]} "
   puts '-----------'
   puts " #{board[6]} | #{board[7]} | #{board[8]} "
 end
 
-def players
+def player_names
   players = []
   2.times do |i|
-    puts "Player #{i + 1}, Please Enter Your Name: "
-    current = gets.chomp.strip
-    puts "Welcome #{current} : All the best"
-    players << current
+    name = ''
+    loop do
+      space
+      puts "player #{i + 1} please Enter Your Name"
+      space
+      name = gets.chomp.upcase
+      break if name.length.positive?
+
+      puts 'Invalid Entry'
+    end
+    puts "Welcome #{name} :"
+    players << name
   end
   players
 end
 
-welcome_message
-display_board(board)
-puts ''
-puts ''
-player_arr = players
-puts "#{player_arr[0]} gets X and #{player_arr[1]} gets O"
-puts 'All the best'
-puts '__' * 70
-display_board(board)
-
-def user_input
-  puts 'Enter Choice between 1 and 9'
-  choice = gets.chomp.strip.to_i
-  choice
+def welcome_players(players)
+  space
+  puts '_' * 50
+  space
+  space
+  puts "                        #{players[0].name}  vs  #{players[1].name}"
+  puts '                    Let the games begin'
+  puts '                       Select from 1-9'
+  puts '_' * 50
+  space
 end
 
-def valid_input?(choice)
-  if choice.is_a?(String) || !choice.between?(1, 9)
-    false
-  else
-    true
-  end
+def end_game
+  puts 'Thank You For Playing Tic Tac Toe'
+  puts 'Game on Vanilla Ruby Build by Gilbert Gotora & Shonibare Adewunmi Comfort'
 end
 
-def free_cell?(board, choice)
-  return false if board[choice - 1] == 'X' || board[choice - 1] == 'O'
-
-  true
+def restart(game, board)
+  space
+  space
+  space
+  puts 'Do You Want To Play Again?'
+  puts 'Enter Y or N'
```

Comment:
- [ ] The game also ends when a user presses something other than 'N', so please fix the instructions.